### ISSUE TITLE
[3.x]: Using a multi-site and GraphQL with relatedToCategories - unexpected entries returned based on field selected in a different site

### ISSUE THREAD

**Issue by ChrisAtRyze** [craftcms/cms#12000]
### Description
Using a multi-site and GraphQL with `relatedToCategories` in the query, entries are returned when they don't have the passed category associated with them and it seems to be because the same entry on another site has that category selected.

### Steps to reproduce

1. Add multiple sites
2. Create a section with a Propagation Method of "Let each entry choose which sites it should be saved to"
3. Create a category group + field for the category (manage relations on a per-site basis) and add it to the entry field layout
4. Create an entry and save it to both sites
5. Create a category and set the category field on the entry on one of the sites
6. Run GraphQL query as below against the site that doesn't have the category selected

#### Query

```
query(
  $categories: CategoryCriteriaInput) {
  projectsList: entries(
    siteId: 2
    section: "Projects"
    relatedToCategories: [$category]
    enabledForSite: true
  ) {
    id
    title
  }
}
```

#### Variables

```
{
  "category": {"slug": "category-slug", "siteId": 2, "enabledForSite": true},
}
```

### Expected behavior

Empty response

### Actual behavior

Returns entry that isn't related to the category

I've tried a few different variations on the query and even just querying the field instead of using related to but coming up with the same issue. Any help would be appreciated.

### Craft CMS version

3.55.2

### PHP version

8.1.7

### Database type and version

MySQL 5.7.33

### Installed plugins and versions

<html>
<body>

Name | Version
-- | --
Amazon S3 | 1.3.0
GeoMate | v1.3.1.2
Redactor | 2.10.10
Super Table | 2.7.3
Tags | 1.0.9

</body>
</html>

**Comment by brandonkelly** [comment]
Thanks for reporting that! The culprit for this is actually how all `relatedTo` params work with element queries. Agree that it’s not the expected behavior and `relatedTo` should only be looking at relations in the queried site(s) by default.

While we can see this being a bug fix, it’s also a minor behavior change, so we didn’t feel comfortable making it in a patch release. Instead we’ve fixed in for the upcoming Craft 4.3 release. (#12072)

**Comment by brandonkelly** [comment]
Craft 4.3 was released earlier this week with that change!
